extension/src/experiments/model/util.ts

```diff
@@ -1,40 +1,38 @@
 import get from 'lodash.get'
 import { formatDate } from '../../util/date'
-import { truncate } from '../../util/string'
 import { splitColumnPath } from '../columns/paths'
 import { Experiment } from '../webview/contract'
 
 type Value = undefined | null | [] | string | number
 
-const getValueType = (value: Value): string => {
-  let type: string = typeof value
+const isDate = (value: Value): boolean =>
+  !!(typeof value === 'string' && Date.parse(value))
+
+const isLongNumber = (value: Value): boolean =>
+  typeof value === 'number' && value.toString().length > 7
+
+const getStringifiedValue = (value: Value): string => {
   if (Number.isNaN(value)) {
-    type = 'NaN'
-  } else if (typeof value === 'string' && Date.parse(value)) {
-    type = 'date'
-  } else if (Array.isArray(value)) {
-    type = 'array'
-  } else if (value === '') {
-    type = 'empty'
+    return 'NaN'
   }
-  return type
-}
 
-const getStringifiedValue = (value: Value): string => {
-  const type = getValueType(value)
-  switch (type) {
-    case 'date':
-      return typeof value === 'string' ? formatDate(value) : ''
-    case 'array':
-      return `[${value?.toString()}]`
-    case 'empty':
-    case 'NaN':
-      return type
-    case 'number':
-      return truncate(String(value), 7)
-    default:
-      return String(value)
+  if (isDate(value)) {
+    return formatDate(value as string)
+  }
+
+  if (Array.isArray(value)) {
+    return `[${value?.toString()}]`
+  }
+
+  if (value === undefined) {
+    return '-'
   }
+
+  if (isLongNumber(value)) {
+    return (value as number).toPrecision(5)
+  }
+
+  return String(value)
```

Comment:
Q: double check what happens with type == empty and NaN - are we getting the same result?

Reply:
`NaN` is the same we are not changing empty string anymore.

Reply:
> we are not changing empty string anymore.

can it break anything downstream by chance?

Reply:
I have double-checked. The only clients of this function are the experiments view tooltip and experiments for plotting quick-pick. Both display the value as expected 👍🏻.